packages/sonarwhal/src/lib/utils/schema-validator.ts

```diff
@@ -0,0 +1,132 @@
+import * as ajv from 'ajv';
+import * as _ from 'lodash';
+
+import { SchemaValidationResult } from '../types';
+
+/*
+ * If we want to use the ajv types in TypeScript, we need to import
+ * ajv in a lowsercase variable 'ajv', otherwise, we can't use types
+ * like `ajv.Ajv'.
+ */
+const validator = new ajv({ // eslint-disable-line new-cap
+    $data: true,
+    allErrors: true,
+    schemaId: 'id',
+    useDefaults: true,
+    verbose: true
+});
+
+validator.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
+
+enum ErrorKeyword {
+    additionalProperties = 'additionalProperties',
+    anyOf = 'anyOf',
+    enum = 'enum',
+    pattern = 'pattern',
+    type = 'type'
+}
+
+const generateError = (type: string, action: ((error: ajv.ErrorObject, property: string, errors?: Array<ajv.ErrorObject>) => string)): ((error: ajv.ErrorObject, errors?: Array<ajv.ErrorObject>) => string) => {
+    return (error: ajv.ErrorObject, errors: Array<ajv.ErrorObject>): string => {
+        if (error.keyword !== type) {
+            return null;
+        }
+
+        const property = error.dataPath.substr(1);
+
+        return action(error, property, errors);
+    };
+};
+
+/**
+ * Returns a readable error for 'additionalProperty' errors.
+ */
+const generateAdditionalPropertiesError = generateError(ErrorKeyword.additionalProperties, (error: ajv.ErrorObject, property: string): string => {
+    const additionalProperty = (error.params as ajv.AdditionalPropertiesParams).additionalProperty;
+
+    return `'${property}' ${error.message}. Additional property found '${additionalProperty}'.`;
```

Comment:
Maybe check if `property` is empty string, and only print the surrounding `'` if it's not?

(Same goes for the others)

Reply:
I don't know if I get what do you want, right now the result is:
if property has value: `'propertyName' message`
if property is empty: `'' message`.

How do you thing the output should be?

Reply:
Yes, right not for an empty property you get something such as: `'' should NOT have additional properties. Additional property found 'z'`. That looks kinda odd, and I was thinking we should drop the `''` part so that the message would be: `'Should NOT have additional properties. Additional property found 'z'`.

Reply:
Ahh, ok!

Reply:
I think this is the only case where property can be an empty string.

Reply:
OK. We will updated things if needed as we use this more. Thanks!